src/imp/security_framework.rs

```diff
@@ -79,17 +94,27 @@ pub struct Pkcs12 {
 
 impl Pkcs12 {
     pub fn from_der(buf: &[u8], pass: &str) -> Result<Pkcs12, Error> {
-        let dir = match TempDir::new("native-tls") {
-            Ok(dir) => dir,
-            Err(_) => return Err(Error(base::Error::from(errSecIO))),
+        SET_AT_EXIT.call_once(|| {
+            extern "C" fn atexit() {
+                TEMP_DIRS.lock().unwrap().clear();
+            }
+            unsafe { libc::atexit(atexit); }
+        });
+
+        let keychain = match TEMP_DIRS.lock().unwrap().entry(buf.into()) {
+            Vacant(entry) => {
+                let dir = TempDir::new("native-tls")
+                    .map_err(|_| Error(base::Error::from(errSecIO)))?;
+                let mut keychain = keychain::CreateOptions::new()
+                    .password(pass)
+                    .create(dir.path().join("tmp.keychain"))?;
+                keychain.set_settings(&KeychainSettings::new())?;
+                entry.insert((keychain.clone(), dir));
+                keychain
+            },
+            Occupied(entry) => entry.get().0.clone(),
         };
 
-        let mut keychain = try!(keychain::CreateOptions::new().password(pass).create(
-            dir.path().join("tmp.keychain"),
-        ));
-        // disable lock on sleep and timeouts
-        try!(keychain.set_settings(&KeychainSettings::new()));
-
         let mut imports = try!(
```

Comment:
Will things behave okay if we import the same identity into the same keychain twice?